simulator/Core.ts

```diff
@@ -43,13 +42,14 @@ export class Core implements ICore {
 
     private triggerEvent(task: ITask, address: number, accessType: CoreAccessType) {
 
-        if (this.pubSubProvider) {
-            this.pubSubProvider.publishSync('CORE_ACCESS', {
+        this.publisher.publish({
```

Comment:
Yeah, like the centralisation of the message types. Would also allow central ways to handle things like sending messages sync/async

Reply:
Yep, that was another motivation for this - from our conversation the other day, where we said we might need to make them async again..